Storage docs

lohanidamodar opened the pull request:
## Preview

![image](https://user-images.githubusercontent.com/6360216/149138860-7392a959-33db-4a87-b482-d9852e4c8794.png)

Light mode

![image](https://user-images.githubusercontent.com/6360216/149138821-ffba275b-ab6b-45ad-9394-bb3df16fc326.png)

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Appwrite storage service is a simple and easy to use service. You can use it to store images, videos, documents, and other files for your projects.</p>
+
+<p>With version 0.13 of Appwrite, we also have introduced storage buckets as a container for your files so that you can give more context to your files. Storage buckets is similar to Collections we have in our <a href="/database">Database</a> service. But a bucket gives you more control over what types of files you allow, what is the maximum file size you allow, whether or not to encrypt the file ...
```
```suggestion
<p>With version 0.13 of Appwrite, we also have introduced storage buckets as a container for your files so that you can give more context to your files. Storage buckets is similar to Collections we have in our <a href="/docs/database">Database</a> service. But a bucket gives you more control over what types of files you allow, what is the maximum file size you allow, whether or not to encrypt the files.</p>
```
Broken link

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Appwrite storage service is a simple and easy to use service. You can use it to store images, videos, documents, and other files for your projects.</p>
+
+<p>With version 0.13 of Appwrite, we also have introduced storage buckets as a container for your files so that you can give more context to your files. Storage buckets is similar to Collections we have in our <a href="/docs/database">Database</a> service. But a bucket gives you more control over what types of files you allow, what is the maximum file size you allow, whether or not to encrypt the ...
```
```suggestion
<p>With version 0.13 of Appwrite, we also have introduced storage buckets as a container for your files so that you can give more context to your files. Storage buckets is similar to Collections we have in our <a href="/docs/database">Database</a> service. The difference is, buckets gives you more control over what types of <b>files</b> you allow, what is the maximum allowed file size, and whether or not to encrypt the files.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h2>Buckets</h2>
+
+<p> Storage buckets are way to group files into a meaningful collection and give them context. Buckets also provide more power to decide what kinds of files, what sizes you want to allow in that bucket. Whether or not to encrypt the files, scan with antivirus and more. Let us look at how we can create a bucket and configure it to our needs.</p>
```
```suggestion
<p> Storage buckets are a way to group files into a meaningful collection and give them context. Buckets also provide more power to decide what kinds of files, what sizes you want to allow in that bucket, whether or not to encrypt the files, scan with antivirus and more. Let's look at how we can create a bucket and configure it to our needs.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+<p>Appwrite storage allow you to manage files in your project. It provides APIs to upload, download, delete, and list files. Not only that Appwrite storage service provides APIs to manipulate image. Using the prveview endpoint you can crop, resize, rotate, add borders and border radius and select the desired output format for your image.</p>
```
```suggestion
<p>Appwrite storage allow you to manage files in your project. It provides APIs to upload, download, delete, and list files. Not only that Appwrite storage service provides APIs to manipulate images. Using the preview endpoint you can crop, resize, rotate, add borders and border radius and select the desired output format for your image.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>You can create your bucket from your Appwrite project's dashboard or using any of our server side SDKs or directly using our REST API authenticated using  an API Key. In order to create a bucket from dashboard, access your Storage from your project's left navigation panel. Click <b>Add Bucket</b> button and choose your bucket's name. For convenience, you can also set a custom ID for your bucke ...
+
+<p>In order to create bucket programmatically, you can either use any of our server side SDKs or REST API paired with an API Key. Let's look at a code sample using CURL, how we can create a bucket using our REST API.</p>
```
```suggestion
<p>In order to create a bucket programmatically, you can either use any of our server side SDKs or REST API paired with an API Key. Let's look at a code sample using CURL, how we can create a bucket using our REST API.</p>
```

Meldiron replied:
We talk about it, but we dont link nor show any examples of way to create bucket using SDK or CURL

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h3>Create Bucket</h3>
+
+<p>You can create your bucket from your Appwrite project's dashboard or using any of our server side SDKs or directly using our REST API authenticated using  an API Key. In order to create a bucket from dashboard, access your Storage from your project's left navigation panel. Click <b>Add Bucket</b> button and choose your bucket's name. For convenience, you can also set a custom ID for your bucke ...
```
```suggestion
<p>You can create your bucket from your Appwrite project's dashboard, using any of our server side SDKs or directly using our REST API authenticated using an API Key. In order to create a bucket from the dashboard, access your Storage from your project's left navigation panel. Click <b>Add Bucket</b> button and choose your bucket's name. For convenience, you can also set a custom ID for your bucket instead of auto-generated ID. This will create a new bucket and take you to its settings page, where you can configure various options for your bucket.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h4>Understanding Bucket Permissions</h4>
+
+As buckets are very similar to Collections we have in database, here again we have similar permission model. Buckets support either bucket level permission or the file level permission to allow different ways to manage access to your buckets and files. Check out the <a href="/docs/permissions">Permissions</a> documentation to learn more about permissions types available.
```
```suggestion
As buckets are very similar to Collections we have in the database, here again, we have a similar permission model. Buckets support either bucket level permission or file level permission to allow different ways to manage access to your buckets and files. Check out the <a href="/docs/permissions">Permissions</a> documentation to learn more about permissions types available.
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h4>Bucket Level Permissions</h4>
+
+<p>With bucket level permission, you only need to provide permissions in your bucket and not in files. If you set the permission to bucket level, file permissions are optional and ignored even if you have set them. This is useful when you want to provide same permission for a collections of files but don't want to repeat the same permission for each file. You provide it once for bucket, set the b ...
```
```suggestion
<p>With bucket level permission, you only need to provide permissions in your bucket and not in files. If you set the permission to bucket level, file permissions are optional and ignored even if you have set them. This is useful when you want to provide the same permission for collections of files but don't want to repeat the same permission for each file. You provide it once for bucket, set the bucket permission to be bucket level and you are done. If you want to update the permission, you only need to do it once in the bucket. With this permission model, if a user has a read and write permission, they will be able to read all the files and modify all the files in that bucket.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>With bucket level permission, you only need to provide permissions in your bucket and not in files. If you set the permission to bucket level, file permissions are optional and ignored even if you have set them. This is useful when you want to provide same permission for a collections of files but don't want to repeat the same permission for each file. You provide it once for bucket, set the b ...
+
+<p>However, with this permission you don't have a granular control over the files inside the bucket. This is more suitable for a use case where you have many people who all have access to same group of files. So instead of adding their permission to each file, you can just create a bucket with bucket level permission and provide them access. Then add all the files they have access to inside that  ...
```
```suggestion
<p>However, with this permission model, you don't have granular control over the files inside the bucket. This model is more suitable for a use case where you have many people who all have access to the same group of files. Instead of adding their permission to each file, you can just create a bucket with bucket level permission and provide them access. Now you can add all the files they should have access to inside that bucket.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h4>File Level Permission</h4>
+
+<p>With file level permission model, you have more granular control over the access of each file. With this permission model, the bucket permissions are ignored and permissions on each file takes into effect whenever we are trying to read or wirte files into the bucket. This is more suitable in the use cases where you want to have different permission for each file. For example you have a bucket  ...
```
```suggestion
<p>With file level permission model, you have more granular control over the access of each file. With this permission model, the bucket permissions are ignored and permissions on each file take into effect whenever we are trying to read or write files into the bucket. This is more suitable in the use cases where you want to have different permission for each file. For example, you have a bucket for user's profile images, and you only want each user to only be able to modify their own image. In such a scenario, a file level permission model will make much more sense and will be easier to manage.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h4>More Bucket Configurations</h4>
+
+<p>Unlike Collections, storage buckets have more configuration options regarding type and size of files that can be in the bucket, encryption and anti-virus scanning. If you look at the bucket settings or the REST API example above you can find these configurations. Let's look at what those are:</p>
```
```suggestion
<p>Unlike Collections, storage buckets have more configuration options regarding the type and size of files that can be in the bucket, encryption and anti-virus scanning. If you look at the bucket settings or the REST API example above you can find these configurations. Let's look at what those are:</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+    <li><code>antivirus</code> parameter allows you to configure whether or not the files being added inside the bucket be scanned by antivirus.</li>
+</ul>
+
+<p>You can learn more about storage buckets and APIs you can use to manage storage buckets from our <a href="docs/server/storage">storage documentation</a>. Now that we know how to create and configure buckets for our applications, we will look at how we can manage files using the storage service next.</p>
```
```suggestion
<p>You can learn more about storage buckets and APIs you can use to manage storage buckets from our <a href="docs/server/storage">Storage documentation</a>. Now that we know how to create and configure buckets for our applications, let's look at how we can manage files using the storage service.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h2>Create File</h2>
+
+<p>Once you create a bucket or have navigated to bucket details, you can access the Files tab from where you can upload, view, delete, update files in the bucket using Appwrite project's dashboard. You can perform all those operations from Appwrite's client and server SDKs as well as REST APIs with proper permission.</p>
```
```suggestion
<p>Once you create a bucket or have navigated to bucket details, you can access the Files tab from where you can upload, view, delete and update files in the bucket using Appwrite project's dashboard. You can also perform all those operations from Appwrite's client SDK, server SDKs and REST APIs as long as you have proper permission.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Once you create a bucket or have navigated to bucket details, you can access the Files tab from where you can upload, view, delete, update files in the bucket using Appwrite project's dashboard. You can perform all those operations from Appwrite's client and server SDKs as well as REST APIs with proper permission.</p>
+
+<p>Once you are in the files tab, you can click <b>Add File</b> and select a file to upload. If the bucket is configured to accept the file type and size you are uploading, your file should be uploaded and you should be able to view the file in the list of files.</p>
```
```suggestion
<p>Once you are in the files tab, you can click <b>Add File</b> and select a file to upload. If the bucket is configured to accept the file type and size you are uploading, your file will be uploaded and you will see the file in the list of files.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Once you are in the files tab, you can click <b>Add File</b> and select a file to upload. If the bucket is configured to accept the file type and size you are uploading, your file should be uploaded and you should be able to view the file in the list of files.</p>
+
+<p>You can also upload files using REST API or SDKs. Below is a CURL syntax to create a file.</p>
```
Missing examples

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h3>Large Files (files above 5MB)</h3>
+
+<p>Above method will only work for files up to 5MB size. If any files are above 5MB, you will need to upload them in chunks. But don't worry, if you are using Appwrite's console or any of our SDKs we handle that internally, so no matter what file size you try to upload, it will work like it worked before. Below is a code example in Dart of how you can chunk and upload each chunk. We are using the ...
```
```suggestion
<p>Above method will only work for files up to 5MB in size. If any files are above 5MB, you will need to upload them in chunks. Don't worry, if you are using Appwrite's console or any of our SDKs we handle that internally, so no matter what file size you try to upload, it will work. Below is a code example in Dart of how you can chunk and upload each chunk. We are using the content-range header provided in the web spec in order to support chunked upload. So if you are using REST API you need to provide chunks and those headers to make a successful upload.</p>
```

Let's drop "as before", because in 5 years people will have no idea what was before

Meldiron replied:
> We are using the content-range header provided in the web spec

Can we link to web specs somewhere?

Meldiron replied:
> Below is a code example in Dart of how you can chunk and upload each chunk.

`chunk and upload chunk` sounds weird, like saying `If you aggregate the file, it will be aggregated.`. Can we rephrase somehow?

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Above method will only work for files up to 5MB size. If any files are above 5MB, you will need to upload them in chunks. But don't worry, if you are using Appwrite's console or any of our SDKs we handle that internally, so no matter what file size you try to upload, it will work like it worked before. Below is a code example in Dart of how you can chunk and upload each chunk. We are using the ...
+
+<!-- dart code example -->
```
Missing code example

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h2>Image Manipulation</h2>
+
+<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite storage's preview endpoint you can manipulate size, add borders and border radius, add background color and set opacity for image and get the image in appropriate output format. This will enable wide range of possibilities from manipulating image to display in responsive websites in appropriate sizes to ...
```
```suggestion
<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite storage's preview endpoint you can manipulate resolution, add borders and border radius, add background colour, set the opacity for the image and get the image in the appropriate output format. This enables a wide range of possibilities! You can manipulate images resolution to display properly on responsive websites. You can also adjust image border, background colour and border radius to match the theming of your application. You can do all of that without caring about how the image was originally uploaded.</p>
```

Last sentence was too long.

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite storage's preview endpoint you can manipulate size, add borders and border radius, add background color and set opacity for image and get the image in appropriate output format. This will enable wide range of possibilities from manipulating image to display in responsive websites in appropriate sizes to ...
+
+<p>Different parameters offered by preview endpoint to manipulate image</p>
```
```suggestion
<p>Below you can find all different parameters offered by preview endpoint to manipulate image</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+</ul>
+
+
+<h2>Download, and streaming with range</h2>
```
```suggestion
<h2>Downloading and streaming with range</h2>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+
+<h2>Download, and streaming with range</h2>
+
+<p>Apart from previewing files and images, downloading is another important feature. And with 0.13 we also support streamed download using the HTTP range header. If you are using our SDKs or console nothing is dirfferent. If you want to download the complete file at once, then even if you use the REST endpoint directly, theresn't any different. However if you want to get only a part of a file fro ...
```
```suggestion
<p>Apart from previewing files and images, downloading is another important feature. With 0.13 we also support streamed download using the HTTP range header. If you are using our SDKs or console, nothing is different. Even if you want to download the whole file at once, even if you use the REST endpoint directly, there isn't any difference. However, if you want to get only a part of a file from storage, you can supply the range header and the server will respond with an appropriate chunk of the file.</p>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,83 @@
+<p>Different parameters offered by preview endpoint to manipulate image</p>
+
+<ul>
+    <li><code>width</code> set the width of the output image, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</li>
+    <li><code>height</code> set the height of the output image, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</li>
+    <li><code>gravity</code> the gravity while cropping the image providing either width or height or both. Accepts any of center, top-left, top, top-right, left, right, bottom-left, bottom, bottom-right</li>
+    <li><code>quality</code> set the quality of the output image. Accepts integer between 0-100, where 100 is the highest quality.</li>
+    <li><code>borderWidth</code> set a border with given width to the output image. Accepts integer between 0-100</li>
+    <li><code>borderColor</code> set a border color for the output image. Accepts any valid Hex color value without the leading #.</li>
+    <li><code>borderRadius</code> set a border radius. Accepts an integer between 0-4000.</li>
+    <li><code>opacity</code> set opacity for the output image. Accepts a float value between 0-1, where 0 makes it transparent. Only works with output format supporting alpha channel like png.</li>
+    <li><code>rotation</code> rotate the output image. Accepts an integer between -360 to 360.</li>
+    <li><code>background</code> set a background color. Accepts any valid Hex color value without the leading #. Only works with output format supporting alpha channel like png.</li>
+    <li><code>output</code> set the output image format. If not provided, will use the original image's format. Supported formats are jpg, jpeg, png, gif and webp.</li>
```
```suggestion
    <li><code>width</code> set the width of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</li>
    <li><code>height</code> set the height of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</li>
    <li><code>gravity</code> the gravity while cropping the image providing either width, height or both. Accepts any of: center, top-left, top, top-right, left, right, bottom-left, bottom, bottom-right</li>
    <li><code>quality</code> set the quality of the output image. Accepts integer between 0-100, where 100 is the highest quality.</li>
    <li><code>borderWidth</code> set a border with given width in pixels to the output image. Accepts integer between 0-100</li>
    <li><code>borderColor</code> set a border color for the output image. Accepts any valid Hex color value without the leading #.</li>
    <li><code>borderRadius</code> set a border radius in pixels. Accepts an integer between 0-4000.</li>
    <li><code>opacity</code> set opacity for the output image. Accepts a float value between 0-1, where 0 makes it transparent. Only works with output format supporting alpha channel like png.</li>
    <li><code>rotation</code> rotate the output image by a degree. Accepts an integer between -360 to 360.</li>
    <li><code>background</code> set a background color. Accepts any valid Hex color value without the leading #. Only works with output format supporting alpha channel like png.</li>
    <li><code>output</code> set the output image format. If not provided, will use the original image's format. Supported formats are: jpg, jpeg, png, gif and webp</li>
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,185 @@
+    -X POST \
+    -H "Content-Type: application/json" \
+    -H "x-appwrite-project: YOUR_PROJECT_ID" \
+    -H "x-appwrite-key: YOUR_KEY_SECRET" \
```
```suggestion
    -H "x-appwrite-key: YOUR_API_KEY" \
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,185 @@
+    -X POST \
+    -H "Content-Type: application/json" \
+    -H "x-appwrite-project: YOUR_PROJECT_ID" \
+    -H "x-appwrite-key: YOUR_KEY_SECRET" \
```
```suggestion
    -H "x-appwrite-key: YOUR_API_KEY" \
```

Meldiron commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,185 @@
+      'content-type': 'multipart/form-data',
+      'content-range': contentRanges,
+      'x-appwrite-project': 'YOUR_PROJECT_ID',
+      'x-appwrite-key': 'YOUR_KEY'
```
```suggestion
      'x-appwrite-key': 'YOUR_API_KEY'
```

Meldiron reviewed (approved):
Looks good to me

eldadfux commented:
A few initial notes about the general structure:

1. The permissions section looks a bit overwhelming. I think we've done a good job on the docs page, would be great to copy-paste-edit and also achieve more consistency.
2. Large-Files I think we should completely abstract away the code example. Although this feature is very important technically it should be transparent to the end user. I think we can add most of the info on how to use it without SDK as a separate blog post, and mention it briefly in the REST spec.
3. Image manipulation -  I think it's great that we have this section. From a glance (didn't read it yet), it looks like we should split it to 2-3 paragraphs it seems now like a huge single one.
4. I think the ending is too dramatic, It could be cool to have a `Learn More` or `Next Steps` section. It might contain only links to the REST client and server docs, not sure yet what else could be beneficial there.

I'll give another more careful review soon.

lohanidamodar commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,297 @@
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+            <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Appwrite } from "appwrite";
+
+const sdk = new Appwrite();
```
Hey, our web SDK doesn't have separate client like other SDKs

brandonroberts replied:
Ok. I was referring to setting the endpoint and project. Is that not required?

eldadfux replied:
@lohanidamodar we should add both project id and the endpoint setting, when the cloud service will be available we can skip the endpoint and rely on the default value.

lohanidamodar commented:
> A few initial notes about the general structure:
> 
> 1. The permissions section looks a bit overwhelming. I think we've done a good job on the docs page, would be great to copy-paste-edit and also achieve more consistency.
> 2. Large-Files I think we should completely abstract away the code example. Although this feature is very important technically it should be transparent to the end user. I think we can add most of the info on how to use it without SDK as a separate blog post, and mention it briefly in the REST spec.
> 3. Image manipulation -  I think it's great that we have this section. From a glance (didn't read it yet), it looks like we should split it to 2-3 paragraphs it seems now like a huge single one.
> 4. I think the ending is too dramatic, It could be cool to have a `Learn More` or `Next Steps` section. It might contain only links to the REST client and server docs, not sure yet what else could be beneficial there.
> 
> I'll give another more careful review soon.

1. Updated with the permission from `docs/database`
2. Agreed, removed and made is part of previous section
3. Doesn't look that big, but have splitted to 2 para and a table of params
4. I am just keeping it consistent with the database docs

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,253 @@
+<p>Appwrite storage allow you to manage files in your project. You can use it to store images, videos, documents, and other files for your projects. It provides APIs to upload, download, delete, and list files. Not only that Appwrite storage service provides APIs to manipulate images. Using the preview endpoint you can crop, resize, rotate, add borders and border radius and select the desired out ...
```
```suggestion
<p>Appwrite storage allow you to manage files in your project. You can use it to store images, videos, documents, and other files for your projects. It provides APIs to upload, download, delete, and list files. Not only that Appwrite storage service provides APIs to manipulate images. Using the preview endpoint you can crop, resize, rotate, add borders and border radius and select the desired output format for your image. The preview API also allows you to change your image's quality, format, or compression, including WebP support for maximum optimization of your network bandwidth.</p>
```

lohanidamodar commented:
@eldadfux latest screenshots
![image](https://user-images.githubusercontent.com/6360216/151923494-8dbc30e8-de5f-4d20-9561-42ae9b70f6c9.png)

![image](https://user-images.githubusercontent.com/6360216/151923457-e2fc0086-2ef4-4415-956a-970755d177b7.png)

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+            <td> parameter allows you to set the maximum size of a file that a bucket accepts. The buckets will accept any file less than or equal to the size provided here.</td>
+        </tr>
+        <tr>
+            <td>encryption</td>
```
Let's add a note about file size limits. We don't encrypt files bigger than X

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+            <td> parameter allows you to configure whether or not the files inside the bucket will be encrypted.</td>
+        </tr>
+        <tr>
+            <td>antivirus</td>
```
Let's add a note about file size limits. We don't scan files bigger than X

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+
+<h2><a href="/docs/storage#createFile" id="createFile">Create File</a></h2>
+
+<p>After you create a bucket or have navigated to bucket details, you can access the Files tab so you can upload, view, delete and update files in the bucket using the Appwrite project's dashboard. You can also perform all those operations from Appwrite's client SDK, server SDKs, and REST APIs as long as you have the proper permission.</p>
```
```suggestion
<p>After you create a bucket or have navigated to bucket details, you can access the `Files` tab so you can upload, view, delete and update files in the bucket using the Appwrite project's dashboard. You can also perform all those operations from Appwrite's client SDK, server SDKs, and REST APIs as long as you have the proper permission.</p>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+
+<p>After you create a bucket or have navigated to bucket details, you can access the Files tab so you can upload, view, delete and update files in the bucket using the Appwrite project's dashboard. You can also perform all those operations from Appwrite's client SDK, server SDKs, and REST APIs as long as you have the proper permission.</p>
+
+<p>When you are in the files tab, you can click <b>Add File</b> and select a file to upload. If the bucket is configured to accept the file type and size you are uploading, your file will be uploaded, and you will see the file in the list of files.
```
```suggestion
<p>When you are in the `Files` tab, you can click <b>Add File</b> and select a file to upload. If the bucket is configured to accept the file type and size you are uploading, your file will be uploaded, and you will see the file in the list of files.
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+    </li>
+</ul>
+
+<p>When you are trying to upload any files above 5MB, you will need to upload them in chunks. Don't worry; if you are using Appwrite's console or any of our SDKs, we handle that internally, so it will work no matter what file size you try to upload.
```
We should add a link to and example of future blog post when that's available.

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+
+<h2><a href="/docs/storage#imagePreview" id="imagePreview">Image Manipulation</a></h2>
+
+<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite storage's <a href="/docs/client/storage#storageGetFilePreview">preview endpoint</a> you can manipulate resolution, add borders and the border radius, add background color, set the opacity for the image, and get the image in the appropriate output format.
```
```suggestion
<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite Storage's <a href="/docs/client/storage#storageGetFilePreview">preview endpoint</a> you can manipulate resolution, add borders and the border radius, add background color, set the opacity for the image, and get the image in the appropriate output format.
```

The Appwrite Storage is the name of our service/product

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+<p>Another great built-in feature of Appwrite is the image manipulation feature. With Appwrite storage's <a href="/docs/client/storage#storageGetFilePreview">preview endpoint</a> you can manipulate resolution, add borders and the border radius, add background color, set the opacity for the image, and get the image in the appropriate output format.
+</p>
+
+<p>This enables a wide range of possibilities! You can manipulate images resolution to display appropriately on responsive websites. You can also adjust the image border, background color, and border radius to match the theming of your application. You can do all that without caring about how the image was originally uploaded.
```
```suggestion
<p>This enables a wide range of possibilities! You can manipulate images resolution to display appropriately on responsive websites. You can also adjust the image border, background color, and border-radius to match the theming of your application. The Appwrite Storage also allows you to change the format and compression of your images for network transfer optimization and to help you speed your application. You can do all that without caring about how the image was originally uploaded.
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+    <tbody>
+        <tr>
+            <td>height</td>
+            <td>set the height of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</td>
```
```suggestion
            <td>Set the height of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>width</td>
+            <td> set the width of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</td>
```
```suggestion
            <td>Set the width of the output image in pixels, the image will be resized keeping the aspect ratio intact. Accepts integer between 0-4000</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>gravity</td>
+            <td> the gravity while cropping the image providing either width, height or both. Accepts any of: center, top-left, top, top-right, left, right, bottom-left, bottom, bottom-right</td>
```
```suggestion
            <td>The gravity while cropping the image providing either width, height or both. Accepts any of: center, top-left, top, top-right, left, right, bottom-left, bottom, bottom-right</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>quality</td>
+            <td> set the quality of the output image. Accepts integer between 0-100, where 100 is the highest quality.</td>
```
```suggestion
            <td>Set the quality of the output image. Accepts integer between 0-100, where 100 is the highest quality.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>borderWidth</td>
+            <td> set a border with given width in pixels to the output image. Accepts integer between 0-100</td>
```
```suggestion
            <td>Set a border with given width in pixels to the output image. Accepts integer between 0-100</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>borderColor</td>
+            <td> set a border color for the output image. Accepts any valid Hex color value without the leading #.</td>
```
```suggestion
            <td>Set a border color for the output image. Accepts any valid Hex color value without the leading #.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>borderRadius</td>
+            <td> set a border radius in pixels. Accepts an integer between 0-4000.</td>
```
```suggestion
            <td>Set a border radius in pixels. Accepts an integer between 0-4000.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>opacity</td>
+            <td> set opacity for the output image. Accepts a float value between 0-1, where 0 makes it transparent. Only works with output format supporting alpha channel like png.</td>
```
```suggestion
            <td>Set opacity for the output image. Accepts a float value between 0-1, where 0 makes it transparent. Only works with output format supporting alpha channel like png.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>rotation</td>
+            <td> rotate the output image by a degree. Accepts an integer between -360 to 360.</td>
```
```suggestion
            <td>Rotate the output image by a degree. Accepts an integer between -360 to 360.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>background</td>
+            <td> set a background color. Accepts any valid Hex color value without the leading #. Only works with output format supporting alpha channel like png.</td>
```
```suggestion
            <td>Set a background color. Accepts any valid Hex color value without the leading #. Only works with output format supporting alpha channel like png.</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+        </tr>
+        <tr>
+            <td>output</td>
+            <td> set the output image format. If not provided, will use the original image's format. Supported formats are: jpg, jpeg, png, gif and webp</td>
```
```suggestion
            <td>Set the output image format. If not provided, will use the original image's format. Supported formats are: jpg, jpeg, png, gif and webp</td>
```

eldadfux commented on app/views/docs/storage.phtml:
```diff
@@ -0,0 +1,256 @@
+
+<h2><a href="/docs/storage#streaming" id="streaming">Downloading and Streaming with Range</a></h2>
+
+<p>Apart from previewing files and images, downloading is another important feature. With 0.13 we also support streamed download using the HTTP range header. If you are using our SDKs or console, nothing is different. Even if you want to download the whole file at once, even if you use the REST endpoint directly, there isn't any difference. However, if you want to get only a part of a file from s ...
```
```suggestion
<p>Apart from previewing files and images, downloading is another important feature. With 0.13 we also support streamed download using the <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range" target="_blank">HTTP range header</a>. If you are using our SDKs or console, nothing is different. Even if you want to download the whole file at once, even if you use the REST endpoint directly, there isn't any difference. However, if you want to get only a part of a file from storage, you can supply the range header and the server will respond with an appropriate chunk of the file.</p>
```
